src/Types.jl

```diff
@@ -1217,7 +999,7 @@ function write_env(ctx::Context; display_diff=true)
         if !ctx.preview
             mkpath(dirname(env.project_file))
             open(env.project_file, "w") do io
-                TOML.print(io, project, sorted=true)
+                TOML.print(io, project, sorted=true, by=project_key_order)
```

Comment:
Maybe sort by `key -> (project_key_order(key), key)`? So that unknown keys are ordered alphabetically.